### PR TITLE
Sorted Orders and KichenOrders listings

### LINKED ISSUE #27: A lista de pedidos deverá retorná-los com suas descrições, ordenados com a seguinte regra:
1. Pronto > Em Preparação > Recebido;

2. Pedidos mais antigos primeiro e mais novos depois;

3. Pedidos com status Finalizado não devem aparecer na lista.

### PR DESCRIPTION
**Pull request by diegobim**
### Changes
- fix typos in `KitchenOrder` related files
- standardize `OrderController` actions
- add `CreatedAt` and `UpdatedAt` fields to `Entity` abstraction and track updates accordingly
- sort `Order` and `KitchenOrder` listings
- add `OrderRepository` integration tests
- add `KitchenOrderRepository` integration tests
- update `KitchenOrder` state machine transitions
- add `KitchenOrder.UpdateStatus` unit tests

---

closes #27

### CODE REVIEW

File: src/SnackHub.Domain/Entities/KitchenOrder.cs [hunk 1]
```diff
@@ -29,13 +29,20 @@ public KitchenOrder(Guid id, Guid orderId, IReadOnlyCollection<KitchenOrdertItem
...
     {
+        var previousStatus = Status;
+        
         Status = Status switch
         {
             KitchenOrderStatus.Received => KitchenOrderStatus.Preparing,
-            KitchenOrderStatus.Preparing => KitchenOrderStatus.Finished,
-            KitchenOrderStatus.Finished => KitchenOrderStatus.Done,
+            KitchenOrderStatus.Preparing => KitchenOrderStatus.Done,
+            KitchenOrderStatus.Done => KitchenOrderStatus.Finished,
```
Comment (diegobim) [inline src/SnackHub.Domain/Entities/KitchenOrder.cs:38]: ⚠️ check with @IcaroSousa if this makes sense: transitions were updated to match the enum order / descriptions

File: src/SnackHub.Infra/Repositories/MongoDB/KitchenOrderRepository.cs [hunk 2]
```diff
@@ -28,6 +29,15 @@ public async Task EditAsync(KitchenOrder kitchenOrder)
...
+        var orders = await ListByPredicateAsync(_ => true);
+        return orders.OrderBy(o => o.CreatedAt);
+    }
+    
+    public async Task<IEnumerable<KitchenOrder>> ListCurrentAsync()
+    {
+        var orders = await ListByPredicateAsync(ko => ko.Status != KitchenOrderStatus.Finished);
+        return orders
+            .OrderByDescending(ko => ko.Status)
+            .ThenBy(ko => ko.CreatedAt);
```
Comment (diegobim) [inline src/SnackHub.Infra/Repositories/MongoDB/KitchenOrderRepository.cs:41]: ℹ️ ideally the sorting should be done by the MongoDB engine, not in memory

File: src/SnackHub.Infra/Repositories/MongoDB/OrderRepository.cs [hunk 3]
```diff
@@ -28,6 +28,7 @@ public async Task EditAsync(Order order)
 
     public async Task<IEnumerable<Order>> ListAllAsync()
     {
-        return await ListByPredicateAsync(order => order.Id != Guid.Empty); // Todo: Add Better Filter, Possible by date
+        var orders = await ListByPredicateAsync(_ => true);
+        return orders.OrderBy(o => o.CreatedAt);
```
Comment (diegobim) [inline src/SnackHub.Infra/Repositories/MongoDB/OrderRepository.cs:32]: ℹ️ ideally the sorting should be done by the MongoDB engine, not in memory

File: src/SnackHub.Infra/Repositories/MongoDB/KitchenOrderRepository.cs (hunk 2 above)
Comment (diegobim) [reply src/SnackHub.Infra/Repositories/MongoDB/KitchenOrderRepository.cs]: fixed

File: src/SnackHub.Infra/Repositories/MongoDB/OrderRepository.cs (hunk 3 above)
Comment (diegobim) [reply src/SnackHub.Infra/Repositories/MongoDB/OrderRepository.cs]: fixed

**Review by sankassio99 — APPROVED** [review]

**Pull request merged**